Source/Selling/Mixins/Commodity.lua

```diff
@@ -0,0 +1,130 @@
+AuctionatorCommoditySellingMixin = {}
+
+local AUCTIONATOR_COMMODITY_EVENTS = {
+  "COMMODITY_SEARCH_RESULTS_UPDATED",
+  "COMMODITY_SEARCH_RESULTS_ADDED"
+}
+
+function AuctionatorCommoditySellingMixin:Initialize()
+  Auctionator.Debug.Message("AuctionatorCommoditySellingMixin:Initialize()")
+
+  hooksecurefunc(
+    AuctionHouseFrame.CommoditiesSellFrame.ItemDisplay,
+    "SetItemInternal",
+    function()
+      self:SetDuration(
+        AuctionHouseFrame.CommoditiesSellFrame.DurationDropDown,
+        tonumber(Auctionator.Config.Get(Auctionator.Config.Options.COMMODITY_AUCTION_DURATION))
+      )
+
+      FrameUtil.RegisterFrameForEvents(self, AUCTIONATOR_COMMODITY_EVENTS)
+    end
+  )
+end
+
+function AuctionatorCommoditySellingMixin:AggregateCommodityResults(itemId)
+  local results = {}
+  local currentResult
+
+  for index = 1, C_AuctionHouse.GetCommoditySearchResultsQuantity(itemId) do
+    currentResult = C_AuctionHouse.GetCommoditySearchResultInfo(itemId, index)
+
+    if currentResult == nil then
+      break
+    end
```

Comment:
When is `currentResult` nil? API quirk?

Reply:
Good question. It was either:
1. Copied from the actual AH code that performs the same function, so preserved because it was necessary there, or
2. An error I ran into while working with it.

It may actually have been a combination of the two. 😄